src/sardana/pool/poolmeasurementgroup.py

```diff
@@ -719,15 +719,13 @@ def set_configuration_from_user(self, cfg):
             # The external controllers should not have synchronizer
 
             if external:
-                if 'synchronizer' in ctrl_data:
-                    raise ValueError('External controller does not allow '
-                                     'to have synchronizer')
-                if 'monitor' in ctrl_data:
-                    raise ValueError('External controller does not allow '
-                                     'to have monitor')
-                if 'timer' in ctrl_data:
-                    raise ValueError('External controller does not allow '
-                                     'to have timer')
+                for parameter in ['synchronizer', 'timer', 'monitor']:
+                    if parameter in ctrl_data:
+                        msg = (
+                            '{} is deprecated for external controllers '
+                            'e.g. Tango, since 3.0.3. Re-apply configuration '
+                            'in order to upgrade.').format(parameter)
+                    self._parent.warning(msg)
```

Comment:
I think line 728 should have one more level of indentation

Reply:
Thanks for pointing this out. I already fixed it.